packages/leafygreen-provider/src/UsingKeyboardProvider.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { createContext, useContext, useState, useMemo } from 'react';
+import PropTypes from 'prop-types';
+import { useEventListener } from '@leafygreen-ui/hooks';
+
+interface UsingKeyboardState {
+  usingKeyboard: boolean;
+  setUsingKeyboard: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+export const UsingKeyboardContext = createContext<UsingKeyboardState>({
+  usingKeyboard: true,
```

Comment:
This should probably default to false, so we don't show a focus state by default everywhere?

Reply:
This is being overly cautious, but this is intentional – the fallback should always be to show focus states first. Once a user clicks anywhere on the page, focus states won't be shown, so this should be a non-issue.

Reply:
Wouldn't we show focus states everywhere then? Wouldn't that cause issues?

If they click anywhere that'll only affect css styles tho right? Not those controlled by this provider?

Reply:
Focus states are still shown with `:focus` CSS pseudo-classes. Where this provider is used, we're simply adding or removing those pseudo-classes rather than toggling focus states on everything.

Reply:
Ah ok, got it. Perfect!

Reply:
Ah ok, got it. Perfect!